scene/gui/graph_edit.h

```diff
@@ -112,12 +112,25 @@ class GraphEdit : public Control {
 	GDCLASS(GraphEdit, Control);
 
 public:
-	struct Connection {
+	struct Connection : RefCounted {
```

Comment:
Is it safe to use `RefCounted` like this without declaring the class with `GDCLASS`? Or registering it?

Reply:
Currently it seems like the only option since we don't allow std::shared_ptr. I'm not a huge fan of this solution either since RefCounted is quite heavy ~352 Bytes and is indeed meant for types which are registered (but since its just an internal struct I don't want to register it). I'd prefer having an equivalent to `std::shared_ptr` for this, but that's out of the scope of this PR.

Reply:
Understand, mainly asked for the actual functionality, if the type isn't registered it isn't guaranteed to work correctly with recounting

Reply:
Just for reference: We discussed this on RocketChat. 
This should be fine since there are other classes/structs which aren't registered and derive from `RefCounted`. You can find some of them using the regex `: public RefCounted.*\n(\t)*^((?!GDCLASS).)*$` (18 results).